core/src/test/java/greencity/controller/HabitStatisticControllerTest.java

```diff
@@ -0,0 +1,262 @@
+package greencity.controller;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import greencity.converters.UserArgumentResolver;
+import greencity.dto.habitstatistic.AddHabitStatisticDto;
+import greencity.dto.habitstatistic.GetHabitStatisticDto;
+import greencity.dto.habitstatistic.HabitItemsAmountStatisticDto;
+import greencity.dto.habitstatistic.HabitStatisticDto;
+import greencity.dto.habitstatistic.UpdateHabitStatisticDto;
+import greencity.dto.user.UserVO;
+import greencity.exception.exceptions.NotFoundException;
+import greencity.exception.handler.CustomExceptionHandler;
+import greencity.service.HabitStatisticService;
+import greencity.service.LanguageService;
+import greencity.service.UserService;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.mockito.junit.jupiter.MockitoSettings;
+import org.mockito.quality.Strictness;
+import org.modelmapper.ModelMapper;
+import org.springframework.boot.web.servlet.error.DefaultErrorAttributes;
+import org.springframework.boot.web.servlet.error.ErrorAttributes;
+import org.springframework.data.web.PageableHandlerMethodArgumentResolver;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+
+import java.security.Principal;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Locale;
+
+import static greencity.ModelUtils.*;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.*;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+@ExtendWith(MockitoExtension.class)
+@MockitoSettings(strictness = Strictness.LENIENT)
+class HabitStatisticControllerTest {
+    private static final String HabitStatisticLink = "/habit/statistic";
+    private MockMvc mockMvc;
+    @Mock
+    private HabitStatisticService habitStatisticService;
+    @Mock
+    private UserService userService;
+    @Mock
+    private ModelMapper modelMapper;
+    @Mock
+    private LanguageService languageService;
+    @InjectMocks
+    private HabitStatisticController habitStatisticController;
+    private ObjectMapper objectMapper;
+    private Principal principal;
+    private UserVO userVO;
+
+    @BeforeEach
+    void setUp() {
+        userVO = getUserVO();
```

Comment:
Why do you re-assign `userVO` before each test?
As far as I see, the method `getUserVO()` returns the same object (object with the same fields everry time, not random) and the code does not mutate it.
Better, once define and use it.